packages/core/src/GenericBanner/GenericBanner.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { Banner } from '../Banner'
+import { Text } from '../Text'
+import { Box } from '../Box'
+import { Flex } from '../Flex'
+import { Link } from '../Link'
+import { Relative } from '../Relative'
+import { Button } from '../Button'
+import { getPaletteColor } from '../utils'
+
+const BannerWithRadius = styled(Banner)`
+  border-radius: 6px;
+  cursor: ${(props) => (props.onClick ? 'pointer' : 'cursor')};
+`
+
+const WrapperLink = styled.a`
+  z-index: 1;
+  display: block;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+`
+
+const CustomLink = styled(Link)`
+  display: inline-flex;
+  z-index: 2;
+  padding: 0;
+  background-color: transparent;
+`
+
+const CustomButton = styled(Button)`
+  z-index: 2;
+  vertical-align: none;
+  text-decoration: underline;
+`
+const GenericBanner = ({
+  alignment,
+  ctaText,
+  heading,
+  iconLeft,
+  iconRight,
+  imageLeft,
+  buttonClick,
+  text,
+  textFontSize,
+  URLProps,
+  ...props
+}) => {
+  return (
+    <Relative>
+      {URLProps && (
+        <WrapperLink tabIndex={-1} aria-hidden='true' {...URLProps} />
+      )}
+      <BannerWithRadius
+        {...props}
+        onClick={URLProps ? null : buttonClick}
+        data-testid='BannerWithRadius'
+      >
+        <Flex alignItems='center' justifyContent={alignment}>
+          {!!iconLeft && iconLeft}
+          {!!imageLeft && imageLeft}
+          <Text px={2} fontSize={textFontSize}>
+            {!!heading && heading}
+            {!!text && text}
+
+            {URLProps && ctaText && (
+              <CustomLink fontSize={textFontSize} {...URLProps}>
+                {!!ctaText && ctaText}
```

Comment:
passed ctaText as a node here, but ideally for accessibility, the node <Text> passed in should have an underline and hover style to let the user know it's an action item. So open to suggestions here. I could change this so it's always an UnderlinedLinkText like below, but also wanted to give more flexibility in what's passed in

```
const UnderlinedLinkText = styled(Text)`
  z-index: 2;
  text-decoration: underline;
  &:hover {
    color: ${(props) => getPaletteColor(props.hoverColor)};
  }
`
```